### PR TITLE
Notify embedded delegate when selection is made

### PR DESCRIPTION
**Pull request by porter-stripe**
## Summary
- Calls `embeddedPaymentElementDidUpdatePaymentOption` when a new selection is made and implements `EmbeddedPaymentElement.paymentOption`.
- Updates the embedded playground to display the current selected payment option and optional mandate
- Adds some new snapshot and unit tests

#### Hides mandate = true

https://github.com/user-attachments/assets/0bd8d6ba-bcd3-47bd-9a5e-60a6838974f3

#### Hides mandate = false

https://github.com/user-attachments/assets/1f4ae94d-22ec-4b0b-8c26-7116931f7b82

## Motivation
- Embedded

## Testing
- Unit tests
- Manual

## Changelog
N/A

### CODE REVIEW

File: StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/Embedded/MandateTextProvider.swift [hunk 1]
```diff
@@ -10,6 +10,8 @@ import Foundation
 @_spi(STP) import StripeUICore
 
 protocol MandateTextProvider {
+    var merchantDisplaysMandate: Bool { get }
```
Comment (yuki-stripe) [inline StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/Embedded/MandateTextProvider.swift:13]: delte this, the computed property in implemntation, and `hidesMandateText` IMO.

File: Example/PaymentSheet Example/PaymentSheet Example/EmbeddedPlaygroundViewController.swift [hunk 2]
```diff
@@ -114,4 +123,93 @@ extension EmbeddedPlaygroundViewController: EmbeddedPaymentElementDelegate {
...
+        setupView()
+    }
+    
+    required init?(coder: NSCoder) {
+        super.init(coder: coder)
+        setupView()
+    }
+    
+    private func setupView() {
+        addSubview(titleLabel)
```
[the remark this answers is not in the document]
Comment (porter-stripe) [reply Example/PaymentSheet Example/PaymentSheet Example/EmbeddedPlaygroundViewController.swift:180]: No strong reason

File: StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/Embedded/EmbeddedPaymentMethodsView.swift [hunk 3]
```diff
@@ -221,3 +240,38 @@ extension PaymentSheet.Appearance.EmbeddedPaymentElement.Style {
...
+        
+        switch selection {
+        case .new(paymentMethodType: let paymentMethodType):
+            image = paymentMethodType.makeImage(
+                forDarkBackground: UITraitCollection.current.isDarkMode,
+                updateHandler: nil
+            )
+            label = paymentMethodType.displayName
+            self.paymentMethodType = paymentMethodType.identifier
+            billingDetails = nil // TODO(porter) Handle billing details when we present forms
```
[the remark this answers is not in the document]
Comment (porter-stripe) [reply StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/Embedded/EmbeddedPaymentMethodsView.swift:259]: Hrm, that's a good flag and I'm not entirely sure. I'm planning to flesh out the billing details stuff when we implement confirming. I'll update the comment for a reminder.

File: StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/Embedded/EmbeddedPaymentMethodsView.swift [hunk 4]
```diff
@@ -182,13 +194,20 @@ class EmbeddedPaymentMethodsView: UIView {
...
+            )
             return
         }
 
         UIView.animate(withDuration: 0.25, animations: {
-            self.mandateView.setHiddenIfNecessary(self.mandateView.attributedText?.string.isEmpty ?? true)
+            self.mandateView.setHiddenIfNecessary(
+                (self.mandateView.attributedText?.string.isEmpty ?? true) ||
+                self.mandateProvider.merchantDisplaysMandate
+            )
```
[the remark this answers is not in the document]
Comment (porter-stripe) [reply StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/Embedded/EmbeddedPaymentMethodsView.swift]: Fixed!

File: StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/Embedded/MandateTextProvider.swift (hunk 1 above)
Comment (porter-stripe) [reply StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/Embedded/MandateTextProvider.swift]: Deleted!

File: StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/Embedded/EmbeddedPaymentElement.swift [hunk 5]
```diff
@@ -38,14 +38,16 @@ public class EmbeddedPaymentElement {
         /// - If this is an external payment method, see https://stripe.com/docs/payments/external-payment-methods?platform=ios#available-external-payment-methods for possible values.
         /// - If this is Apple Pay, the value is "apple_pay"
         public let paymentMethodType: String
-        /// If you set `configuration.hidesMandateText = true`, this text must be displayed to the customer near your “Buy” button to comply with regulations.
+        /// If you set `configuration.embeddedViewDisplaysMandateText = true`, this text must be displayed to the customer near your “Buy” button to comply with regulations.
```
Comment (yuki-stripe) [inline StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/Embedded/EmbeddedPaymentElement.swift:41]:
```suggestion
        /// If you set `configuration.embeddedViewDisplaysMandateText = false`, this text must be displayed to the customer near your “Buy” button to comply with regulations.
```

File: StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/Embedded/EmbeddedPaymentMethodsView.swift [hunk 6]
```diff
@@ -54,10 +55,12 @@ class EmbeddedPaymentMethodsView: UIView {
          shouldShowApplePay: Bool,
          shouldShowLink: Bool,
          savedPaymentMethodAccessoryType: RowButton.RightAccessoryButton.AccessoryType?,
-         mandateProvider: MandateTextProvider) {
+         mandateProvider: MandateTextProvider,
+         shouldShowMandate: Bool = true) {
         self.appearance = appearance
         self.selection = initialSelection
         self.mandateProvider = mandateProvider
+        self.shouldShowMandate = shouldShowMandate
```
Comment (yuki-stripe) [inline StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/Embedded/EmbeddedPaymentMethodsView.swift:63]: I'd think we're going to pass this thing a `Configuration` eventually, so we'll probably get rid of this and just reach directly for `configuration.embeddedViewDisplaysMandateText` but we can change it when we get there.

**Review by yuki-stripe — APPROVED** [review]

**Review by yuki-stripe — APPROVED** [review]

**Pull request merged**